### ISSUE TITLE
Formatting issue when using this library in a form

### ISSUE THREAD

**Issue by RubenWaterman** [nkezhaya/live_phone#87]
First of all, thanks a lot for putting this library together!

I couldn't find documentation on using this inside a Phoenix (Liveview) form. Still, I adapted the example to do it like this (I'm on Phoenix Liveview 0.18, so there is no longer the need to import the socket on a live component):
```
<%= live_component(LivePhone.Component, id: "dp_mobile_phone", form: f, field: :dp_mobile_phone, apply_format?: true, preferred: ["US", "GB", "CA"]) %>
```

That seems to work alright when the form is used for the first time:
![Screenshot 2022-12-12 at 16 48 40](https://user-images.githubusercontent.com/5938863/207105286-d78d7abf-dd28-4ac6-95f3-1a51542f20e1.png)

However, when I returned to the form for a second time, the country was set to the US again. I expect the country to be set based on the stored country code and not show the +31 (in this example) in the input field. Am I missing some initialization in the above code, or has this not come up before?

![Screenshot 2022-12-12 at 16 48 54](https://user-images.githubusercontent.com/5938863/207105895-7bd2a556-ea51-4936-94e8-51419154e304.png)

Furthermore, would it make sense to be able to pass (optional) parameters for the CSS class on the component so I can easily make this phone field look the same as my other inputs?

**Comment by nkezhaya** [comment]
>However, when I returned to the form for a second time, the country was set to the US again.

@RubenWaterman Do you mean that you're loading a new page with a phone number already present?

**Comment by nkezhaya** [comment]
I wonder if this is a new bug caused by #83

**Comment by RubenWaterman** [comment]
> > However, when I returned to the form for a second time, the country was set to the US again.
> 
> @RubenWaterman Do you mean that you're loading a new page with a phone number already present?

Correct

**Comment by nkezhaya** [comment]
Working on reproducing this now.

**Comment by nkezhaya** [comment]
@RubenWaterman I can definitely reproduce this! It's been a long time since I worked on this or used it, so my memory is a bit foggy. I think we never really added full support for passing the form object you would get from `<.form :let={f} ...>`.

I don't remember what we did, because as a _user_ I definitely needed it for a form representing an Ecto changeset, but it might have been something like: `<.live_component module={LivePhone.Component} value={input_value(f, :phone)} />`

@maxvw does this ring a bell?

**Comment by maxvw** [comment]
Not really, it's been a while for me too. Looking at an old commit in the project it was used does show it being used with `form: f, field: :phone` so I imagine it did work at some point. But I think it might be related to the [`apply_format?`][1] option, I'm not sure we ever ended up using that ourselves. We also parsed and used the `e164` format in our changeset, not sure if that maybe makes a difference here. And of course we only used it with older LV versions but I'm not sure if that should matter here.

Don't really have time right now to play with this myself so hopefully this was a useful comment.

[1]: https://github.com/nkezhaya/live_phone/pull/4

**Comment by RubenWaterman** [comment]
@maxvw / @nkezhaya did you ever have two of these form inputs in the same form? I don't seem to be able to get this to work. Even though they have separate IDs, if I make a change in one phone field, it updates both form fields.

```
<%= live_component(LivePhone.Component, id: "dp_phone", form: f, field: :dp_phone, apply_format?: true, preferred: get_preferred_countries(input_value(f, :dp_phone))) %>
<%= live_component(LivePhone.Component, id: "dp_mobile_phone", form: f, field: :dp_mobile_phone, apply_format?: true, preferred: get_preferred_countries(input_value(f, :dp_mobile_phone))) %>
```

So let's say I have my form like this (I only edited dp_mobile_phone field):

![Screenshot 2022-12-13 at 10 33 26](https://user-images.githubusercontent.com/5938863/207294646-ae8b5d7a-b0d2-4fd9-9a7c-dca488ea061b.png)

And then click "Save", somehow both phone fields are then updated to the new number:
![Screenshot 2022-12-13 at 10 34 21](https://user-images.githubusercontent.com/5938863/207294820-03d3fdd7-fd6e-4393-a387-6a71884ec4eb.png)

**Comment by nkezhaya** [comment]
> did you ever have two of these form inputs in the same form? I don't seem to be able to get this to work. Even though they have separate IDs, if I make a change in one phone field, it updates both form fields.

Never tried that! I'm going to try updating the example app today with something more complex and see what I can come up with.

**Comment by RubenWaterman** [comment]
> Never tried that! I'm going to try updating the example app today with something more complex and see what I can come up with.

Nice! bear in mind that if you have a `phx-change="validate"` on the form, you cannot even make any changes to either phone field as you would be trying to update a form field with conflicting values (from the two phone fields). I got around that by removing the `phx-change="validate"` from the whole field and add it on the individual form tags of the other fields.

**Comment by nkezhaya** [comment]
@RubenWaterman Try now using the `dev` branch and let me know if that's any improvement?

**Comment by RubenWaterman** [comment]
Hey! Thanks a lot! It works now (while having `phx-change="validate"`) on the entire form). However, the components don't seem to be completely isolated, as typing in one field affects the validator of the other form. Any ideas on why that happens?

![phx-liveview](https://user-images.githubusercontent.com/5938863/207405981-81ed9e69-f99f-4e0e-8712-367d1851f1de.gif)

**Comment by nkezhaya** [comment]
@RubenWaterman I'll have time to look into this tomorrow, but I think I have an idea

**Comment by nkezhaya** [comment]
@RubenWaterman Curious to see if this is working for you now? Just pushed another update to `dev`.

**Comment by RubenWaterman** [comment]
> @RubenWaterman Any luck?

Hey! I just installed your latest version, but unfortunately, that broke something 👀  As soon as the page is loaded, events start firing non-stop from all the live_phone input fields. The page becomes completely unresponsive (to any other event or even the country drop-down).

**Comment by nkezhaya** [comment]
Could you publish that repo somewhere?

**Comment by RubenWaterman** [comment]
> Could you publish that repo somewhere?

The project is quite extensive with a lot of bits and pieces that require setup, so I wanted to make a little repo to mimic the issue but I haven't gotten to it yet :( I will post back here asap with an example repo.

**Comment by nkezhaya** [comment]
@RubenWaterman I think I've gotten LivePhone to a pretty decent place and did a lot of refactoring. I still wasn't able to reproduce the bug you encountered, so I was going to merge the `dev` branch soon and publish a new release.
